pkg/operator/internal/webhooks/catapult_webhook.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	adminv1beta1 "k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	operatorv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/operator/v1alpha1"
+	"github.com/kubermatic/kubecarrier/pkg/internal/util/webhook"
+)
+
+// CatapultWebhookHandler handles validating of Catapult objects.
+type CatapultWebhookHandler struct {
+	decoder *admission.Decoder
+	Log     logr.Logger
+}
+
+var _ admission.Handler = (*CatapultWebhookHandler)(nil)
+
+// +kubebuilder:webhook:path=/mutate-operator-kubecarrier-io-v1alpha1-catapult,mutating=true,failurePolicy=fail,groups=operator.kubecarrier.io,resources=catapults,verbs=create,versions=v1alpha1,name=mcatapult.kubecarrier.io
+
+// Handle is the function to handle create requests of Catapults.
+func (r *CatapultWebhookHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	obj := &operatorv1alpha1.Catapult{}
+	if err := r.decoder.Decode(req, obj); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	switch req.Operation {
+	case adminv1beta1.Create:
+		r.Log.Info("creating catapult", "spec", obj.Spec)
+		if obj.Spec.LogLevel == nil {
+			webhook.SetDefaultLogLevel(webhook.LogLevelSetter(&obj.Spec))
```

Comment:
Why not fetch `kubecarrier` object directly and set its log value as default one? The object would already be in a cache, thus it's no network calls. Furthermore, this would negate the need for a deep global variable fetch.

Reply:
doen